torch/csrc/autograd/FunctionsManual.h

```diff
@@ -32,6 +32,9 @@ struct IndexRangeGenerator {
     size_t i = 0;
 };
 
+inline bool isFwGradDefined(const Tensor& t) {
```

Comment:
Why is this added? The overload below already covers that no?
Also it would be nice to reduce code in header files...

Reply:
The issue is that with just the optional overload we wrap `t` (due to the implicit conversion), and then immediately check that it is defined and then unwrap it on the other side. That turns out to be a non-trivial amount of added work. (Same with inlining; it's a small function on the hot path so we want it to be inlined.)